src/hooks/shards/useShardEndpoints.ts

```diff
@@ -1,25 +1,24 @@
-import useScopedGatewayAuthToken from 'hooks/useScopedGatewayAuthToken';
 import { useMemo } from 'react';
 import { useShardDetail_readDictionary } from 'stores/ShardDetail/hooks';
 import { ShardEntityTypes } from 'stores/ShardDetail/types';
+import { hasLength } from 'utils/misc-utils';
 
 export const useShardEndpoints = (
     taskName: string,
-    taskTypes: ShardEntityTypes[]
+    taskTypes: ShardEntityTypes[],
+    reactorAddress: string | undefined
 ) => {
     const dictionaryVal = useShardDetail_readDictionary(taskName, taskTypes);
 
-    const gateway = useScopedGatewayAuthToken(taskName);
     const gatewayHostname = useMemo(() => {
-        if (gateway.data?.gateway_url) {
-            // Even though `gateway_url` is already a `URL` object, the
-            // `host` property returns `null` for some $jsReason
-            const url = new URL(gateway.data.gateway_url.toString());
+        if (typeof reactorAddress === 'string' && hasLength(reactorAddress)) {
+            const url = new URL(reactorAddress);
```

Comment:
Since this is a bit more dynamic now... we should probably catch the error now if the `reactorAddress` isn't a valid URL.

Reply:
I think we could probably just return null and log a LR event for now.